autoload/coc/powershell/utils.vim

```
@@ -0,0 +1,80 @@
+
+function! s:getAvailablePowerShellExecutables ()
+    let paths = []
+    if(s:is_win)
+        call add(paths, {
+            \ versionName: "Windows PowerShell",
+            \ executablePath: "C:\\System32\\WindowsPowerShell\\v1.0\\powershell.exe"
+            \ })
+        
+        let psCoreInstallPath = "C:\\System32\\PowerShell\\"
+        if(isdirectory(psCoreInstallPath))
+            let psCoreExePaths = split(glob(psCoreInstallPath . "**\\pwsh.exe"), "\n")
+            if(!empty(psCoreExePaths))
+                call add(paths, {
+                    \ versionName: "PowerShell Core",
+                    \ executablePath: psCoreExePaths[0]
+                    \ })
+            endif
+        endif
+
+        return paths
+    endif
+
+    " macOS and Linux
+    let powershellExecutables = [ s:linuxExe, s:linuxPreviewExe, s:snapExe, s:snapPreviewExe ]
+    if(s:is_mac)
+        let powershellExecutables = [ s:macOSExe, s:macOSPreviewExe ]
+    endif
+
+    for powershellExecutable in powershellExecutables
+        if(filereadable(powershellExecutable.executablePath))
+            call add(paths, powershellExecutable)
+        endif
+    endfor
+
+    return paths
+endfunction
+
+function! coc#powershell#utils#switch_powershell_executable (...)
+    let s:choice = ""
+    if(!exists("a:1"))
+        let powershellExecutables = s:getAvailablePowerShellExecutables()
+        let index = 1
+        let choiceStr = ""
+        while(index <= len(powershellExecutables))
+            let choiceStr = choiceStr . index . " " . powershellExecutables[index - 1].versionName
+            if(index != len(powershellExecutables))
+                let choiceStr = choiceStr . "\n"
+            endif
+            let index = index + 1
+        endwhile
+        let s:choice = powershellExecutables[confirm("Which PowerShell executable would you like to use?", choiceStr) - 1].executablePath
```

`help confirm()`: 		If the user aborts the dialog by pressing <Esc>, CTRL-C, or another valid interrupt key, confirm() returns 0.

---

fixed - now it silently returns if you cancel the prompt.